sql/parse/parse.go

```diff
@@ -2852,9 +2864,21 @@ func StringToColumnDefaultValue(ctx *sql.Context, exprStr string) (*sql.ColumnDe
 	if err != nil {
 		return nil, err
 	}
-	// The literal and expression distinction seems to be decided by the presence of parentheses, even for defaults like NOW() vs (NOW())
+	// The literal and function expression distinction seems to be decided by the presence of parentheses, even for defaults like NOW() vs (NOW())
 	// 2+2 would evaluate to a literal under the parentheses check, but will have children due to being an Arithmetic expression, thus we check for children.
-	return ExpressionToColumnDefaultValue(ctx, parsedExpr, len(parsedExpr.Children()) == 0 && !strings.HasPrefix(exprStr, "("))
+
+	var isLiteral bool
+	switch e := parsedExpr.(type) {
+	case *expression.UnaryMinus:
+		_, isLiteral = e.Child.(*expression.Literal)
+	case *expression.UnresolvedFunction:
+		if !strings.HasPrefix(exprStr, "(") && !strings.HasPrefix(exprStr, ")") && (e.Name() == "now" || e.Name() == "current_timestamp") {
```

Comment:
(minor) do we need the closing `)` paren if? I couldn't see why that would be a prefix.